src/metatrain/utils/llpr.py

```diff
@@ -61,22 +63,24 @@ def __init__(
         )
 
         # register covariance and inverse covariance buffers
-        self.register_buffer(
-            "covariance",
-            torch.zeros(
+        device = next(self.model.parameters()).device
```

Comment:
Since these are no longer buffers, will they be automatically moved to new dtype/devices?

Reply:
Alternatively, the API could change here as well to explicitly specify which output should we apply LLPR to. I feel like this would be a lot cleaner & more composable.

Reply:
They are sent to the correct device in the forward, the dtype doesn't change (corresponds to that of the model). I didn't find a better way, since torch doesn't support dicts of buffers. This is the same that we do with labels